internal/api/core/routing.go

```diff
@@ -146,10 +150,16 @@ func sendAPIResponse(apiRequest ValidAPIRequest, response http.ResponseWriter,
 
 			payload, _ = util.ToJSON(apiResponse)
 		} else {
-			return sendAPIResponse(apiRequest, response, nil, apiErr, true)
+			return sendAPIResponse(apiRequest, response, nil, apiErr, true, startTime, sender)
 		}
 	}
 
+	// Don't log request stats when in testing mode.
```

Comment:
Why disable the stat collection?
All our other stats log during testing.
https://github.com/edulinq/autograder-server/blob/main/internal/grader/grade.go#L116

They may do it synchronously, but they still collect.